nixos/modules/services/backup/restic.nix

```diff
@@ -256,6 +256,24 @@ in
             having to manually specify most options.
           '';
         };
+
+        niceness = mkOption {
+          description = lib.mdDoc "Niceness for instances of restic. Also applies to remote ones connecting via ssh when positive. Only used for `restic backup`, `restic forget` and `restic check` commands.";
+          type = types.ints.between (-20) 19;
+          default = 10;
+        };
+
+        ioSchedulingClass = mkOption {
+          description = lib.mdDoc "IO scheduling class for restic (see ionice(1) for a quick description). Applies to local instances, and remote ones connecting by ssh if set to idle. Only used for `restic backup`, `restic forget` and `restic check` commands.";
+          type = types.enum [ "idle" "best-effort" "realtime" ];
+          default = "best-effort";
+        };
+
+        ioPriority = mkOption {
+          description = lib.mdDoc "IO scheduling priority for restic (see ionice(1) for a quick description). If null, the value is derived from (niceness + 20) / 5. Only used for `restic backup`, `restic forget` and `restic check` commands.";
+          type = types.nullOr (types.ints.between 0 7);
```

Comment:
[ionice(1)](https://linux.die.net/man/1/ionice) states that the "idle" schedule class does not take a priority argument, and only "best effort" and "real time" expect an int within 0-7. Thus, either reflect this restriction with a check or at least add it to the description.